src/core/service.h

```diff
@@ -116,6 +116,9 @@ struct Service {
         char *pid_file;
 
         usec_t restart_usec;
+        unsigned restart_steps;
+        usec_t restart_usec_max;
+        usec_t restart_usec_cur;
```

Comment:
why store this btw? why no calculate this on the fly whenever we need it?

Reply:
We need the value before and after `n_restarts` is updated, which will require extra handling in that way. So I just made the conclusion that it is better to store it 🤔

Reply:
Also I find it useful to expose the actual time we'd be waiting through dbus. Storing it might be more efficient?

Reply:
Hmm, don't mind me. I reevaluated this and now I think the other way is better